src/Program.cs

```diff
@@ -30,61 +26,48 @@ public static async Task Main(string[] args)
             using (var serviceScope = host.Services.CreateScope())
             {
                 var services = serviceScope.ServiceProvider;
-                await InternalMain(args, services.GetRequiredService<PackageHelper>());
-                   
-            }
-        }
-
-        private static async Task InternalMain(string[] args, PackageHelper packageHelper)
-        {
+                var packageHelper = services.GetRequiredService<PackageHelper>();
 
-            // now uses 'verbs' so each verb is a command
-            // 
-            // e.g umbpack init or umbpack push
-            //
-            // these are handled by the Command classes.
+                using var parser = new Parser(with =>
+                {
+                    with.HelpWriter = null;
+                    with.AutoVersion = false;
+                    with.CaseSensitive = false;
+                });
 
-            var parser = new Parser(with => {
-                with.HelpWriter = null;
-                // with.HelpWriter = Console.Out;
-                with.AutoVersion = false;
-                with.CaseSensitive = false;
-            } );
+                var parserResult = parser.ParseArguments<PackOptions, PushOptions, InitOptions>(args);
 
-            // TODO: could load the verbs by interface or class
-
-            var parserResults = parser.ParseArguments<PackOptions, PushOptions, InitOptions>(args);
-
-            parserResults
-                .WithParsed<PackOptions>(opts => PackCommand.RunAndReturn(opts).Wait())
-                .WithParsed<PushOptions>(opts => PushCommand.RunAndReturn(opts, packageHelper).Wait())
-                .WithParsed<InitOptions>(opts => InitCommand.RunAndReturn(opts))
-                .WithNotParsed(async errs => await DisplayHelp(parserResults, errs));
+                return (int)await parserResult.MapResult(
```

Comment:
This maps the results from the different commands (that return the `ErrorCode` enum) to the return/exit code.